mp/src/game/shared/momentum/weapon/weapon_mom_stickybomblauncher.cpp

```diff
@@ -0,0 +1,424 @@
+#endif
+
+    CMomentumPlayer *pPlayer = GetPlayerOwner();
+
+    if (!pPlayer)
+        return;
+
+    // Are we capable of firing again?
+    if (m_flNextPrimaryAttack > gpGlobals->curtime)
+        return;
+
+    if (m_flChargeBeginTime <= 0 && mom_sj_charge_enable.GetBool())
+    {
+        // save that we had the attack button down
+        m_flChargeBeginTime = gpGlobals->curtime;
+
+        if (m_bIsChargeEnabled.Get())
+        {
+            SendWeaponAnim(ACT_VM_PULLBACK);
+            WeaponSound(GetWeaponSound("charge"));
+        }
+    }
+    else
+    {
+        float flTotalChargeTime = gpGlobals->curtime - m_flChargeBeginTime;
+
+        if (flTotalChargeTime >= MOM_STICKYBOMB_MAX_CHARGE_TIME)
+        {
+            LaunchGrenade();
+        }
+    }
+}
+
+void CMomentumStickybombLauncher::SecondaryAttack()
+{
+    if (m_iStickybombCount == 0)
+        return;
+
+    const auto pPlayer = GetPlayerOwner();
+
+    if (!pPlayer)
+        return;
+
+    if (m_iStickybombCount)
+    {
+        if (DetonateRemoteStickybombs(false))
+        {
+            if (m_flLastDenySoundTime <= gpGlobals->curtime)
+            {
+                // Deny!
+                m_flLastDenySoundTime = gpGlobals->curtime + 1;
+                WeaponSound(GetWeaponSound("deny"));
+            }
+        }
+        else
+        {
+            // Play a detonate sound.
+            WeaponSound(GetWeaponSound("detonate"));
+        }
+    }
+}
+
+float CMomentumStickybombLauncher::GetProjectileSpeed()
+{
+#ifdef CLIENT_DLL
+    static ConVarRef mom_sj_charge_enable("mom_sj_charge_enable");
+#endif
+
+    if (!mom_sj_charge_enable.GetBool())
+        return 900.0f;
+
+    return RemapValClamped((gpGlobals->curtime - m_flChargeBeginTime), 0.0f, 
+                           MOM_STICKYBOMB_MAX_CHARGE_TIME, 
+                           MOM_STICKYBOMB_MIN_CHARGE_VEL, 
+                           MOM_STICKYBOMB_MAX_CHARGE_VEL);
+}
+
+
+//-----------------------------------------------------------------------------
+// Purpose: Add stickybombs to our list as they're fired
+//-----------------------------------------------------------------------------
+CMomStickybomb *CMomentumStickybombLauncher::FireProjectile(CMomentumPlayer *pPlayer)
+{
+    const auto pProjectile = FireStickybomb(pPlayer);
+#ifdef GAME_DLL
+    if (pProjectile)
+    {
+        // If we've gone over the max stickybomb count, fizzle the oldest
+        if (m_Stickybombs.Count() >= MOM_WEAPON_STICKYBOMB_COUNT)
+        {
+            CMomStickybomb *pTemp = m_Stickybombs[0];
+            if (pTemp)
+            {
+                pTemp->Fizzle();
+                pTemp->RemoveStickybomb(true);
+            }
+
+            m_Stickybombs.Remove(0);
+        }
+
+        CMomStickybomb *pStickybomb = pProjectile;
+        pStickybomb->SetLauncher(this);
+
+        StickybombHandle hHandle;
+        hHandle = pStickybomb;
+        m_Stickybombs.AddToTail(hHandle);
```

Comment:
You could do `AddStickybomb(pProjectile);` here instead of these lines; it's the same code.

Either that or remove `AddStickybomb` if it's unused.